### ISSUE TITLE
v0.7 compatability

### ISSUE THREAD

**Issue by ZacLN** [JuliaLang/Tokenize.jl#123]
A tracker issue for changes that will need to be made for v0.7

- [ ] #22089: allow suffixed operators in Tokenize
- [x] #24404: add operator to Tokenize
- [x]  introduce specific literal types for bin/hex/oct to Tokenize

**Comment by JeffreySarnoff** [comment]
with v0.7.0-
```
julia> using Tokenize
[ Info: Precompiling module Tokenize
WARNING: could not import Base.UTF8proc into Lexers
ERROR: LoadError: LoadError: LoadError: Base.CodePointError(0xffffffff)
Stacktrace:
 [1] code_point_err(::UInt32) at ./char.jl:10
```

**Comment by KristofferC** [comment]
Check out the package.

**Comment by JeffreySarnoff** [comment]
yes -- I checked out the package first (retried):
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.4020 (2018-02-19 22:10 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 1bbd518* (1 day old master)
|__/                   |  x86_64-linux-gnu

julia> Pkg.checkout("Tokenize")
WARNING: Base.Pkg is deprecated, run `using Pkg` instead
 in module Main
[ Info: Checking out Tokenize master...
[ Info: Pulling Tokenize latest master...
┌ Warning: Nullables is fixed at 0.0.1 conflicting with requirement for TimeZones: [0.0.3,∞)
└ @ Pkg.Query query.jl:30
┌ Warning: Nemo is fixed at 0.0.0- conflicting with requirement for ArbFloats: [0.7.0,∞)
└ @ Pkg.Query query.jl:30
ERROR: Unsatisfiable requirements detected for package CUDAdrv:
├─version range [0.0.0-,∞) set by an explicit requirement
├─version range [0.7.0,∞) required by package CuArrays, whose only allowed version is 0.4.0+:
│ ├─version 0.4.0+ set by fixed requirement (package is checked out, dirty or pinned)
│ └─version range [0.0.0-,∞) set by an explicit requirement
└─version range [0.2.4,0.7.0) required by package CUDArt, whose allowed version range is [0.3.0,∞):
  └─version range [0.3.0,∞) required by package CUSPARSE, whose only allowed version is 0.6.0+:
    ├─version 0.6.0+ set by fixed requirement (package is checked out, dirty or pinned)
    └─version range [0.0.0-,∞) set by an explicit requirement
The intersection of the requirements is empty.
Stacktrace:
 [1] filter_versions(::Dict{String,Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Pkg.Types.Available}}, ::Dict{AbstractString,Pkg.Types.ResolveBacktraceItem}) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/query.jl:343
 [2] prune_versions(::Dict{String,Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Pkg.Types.Available}}, ::Dict{AbstractString,Pkg.Types.ResolveBacktraceItem}) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/query.jl:372
 [3] prune_dependencies at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/query.jl:586 [inlined]
 [4] resolve(::Dict{String,Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Pkg.Types.Fixed}, ::Dict{String,VersionNumber}, ::Set{String}) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/entry.jl:508
 [5] resolve at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/entry.jl:487 [inlined] (repeats 6 times)
 [6] (::getfield(Pkg.Entry, Symbol("##16#18")){String,String,Bool,Bool})(::LibGit2.GitRepo) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/entry.jl:245
 [7] transact(::getfield(Pkg.Entry, Symbol("##16#18")){String,String,Bool,Bool}, ::LibGit2.GitRepo) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/LibGit2/src/LibGit2.jl:910
 [8] #15 at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/entry.jl:236 [inlined]
 [9] with(::getfield(Pkg.Entry, Symbol("##15#17")){String,String,Bool,Bool}, ::LibGit2.GitRepo) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/LibGit2/src/types.jl:1067
 [10] with at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/LibGit2/src/types.jl:1073 [inlined]
 [11] checkout(::String, ::String, ::Bool, ::Bool) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/entry.jl:235
 [12] #3 at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/dir.jl:33 [inlined]
 [13] cd(::getfield(Pkg.Dir, Symbol("##3#6")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},typeof(Pkg.Entry.checkout),Tuple{String,String,Bool,Bool}}, ::String) at ./file.jl:70
 [14] #2 at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/dir.jl:33 [inlined]
 [15] withenv(::getfield(Pkg.Dir, Symbol("##2#5")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},typeof(Pkg.Entry.checkout),Tuple{String,String,Bool,Bool},String}, ::Pair{String,String}, ::Vararg{Pair{String,String},N} where N) at ./env.jl:156
 [16] #cd#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/dir.jl:32
 [17] cd at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/dir.jl:25 [inlined]
 [18] #checkout#1(::Bool, ::Bool, ::Function, ::String, ::String) at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/Pkg.jl:189
 [19] checkout at /home/jas/Public/julia/julia/usr/share/julia/site/v0.7/Pkg/src/Pkg.jl:189 [inlined] (repeats 2 times)
 [20] top-level scope

julia> collect(tokenize("function f(x) end"))
ERROR: UndefVarError: tokenize not defined

julia> using Tokenize
[ Info: Precompiling module Tokenize
WARNING: could not import Base.UTF8proc into Lexers
ERROR: LoadError: LoadError: LoadError: Base.CodePointError(0xffffffff)
Stacktrace:
 [1] code_point_err(::UInt32) at ./char.jl:10
 [2] Type at ./char.jl:54 [inlined]
 [3] convert(::Type{Char}, ::UInt32) at ./char.jl:72
 [4] top-level scope
 [5] include at ./boot.jl:295 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1064
 [7] include at ./sysimg.jl:27 [inlined]
 [8] include(::String) at /home/jas/.julia/v0.7/Tokenize/src/lexer.jl:1
 [9] top-level scope
 [10] include at ./boot.jl:295 [inlined]
 [11] include_relative(::Module, ::String) at ./loading.jl:1064
 [12] include at ./sysimg.jl:27 [inlined]
 [13] include(::String) at /home/jas/.julia/v0.7/Tokenize/src/Tokenize.jl:3
 [14] top-level scope
 [15] include at ./boot.jl:295 [inlined]
 [16] include_relative(::Module, ::String) at ./loading.jl:1064
 [17] include(::Module, ::String) at ./sysimg.jl:27
 [18] top-level scope
 [19] eval at ./boot.jl:298 [inlined]
 [20] top-level scope at ./<missing>:3
in expression starting at /home/jas/.julia/v0.7/Tokenize/src/utilities.jl:31
in expression starting at /home/jas/.julia/v0.7/Tokenize/src/lexer.jl:3
in expression starting at /home/jas/.julia/v0.7/Tokenize/src/Tokenize.jl:6
ERROR: Failed to precompile Tokenize to /home/jas/.julia/lib/v0.7/Tokenize.ji.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] compilecache(::Base.PkgId) at ./loading.jl:1202
 [3] _require(::Base.PkgId) at ./loading.jl:1000
 [4] require(::Base.PkgId) at ./loading.jl:871
 [5] require(::Module, ::Symbol) at ./loading.jl:866
```

**Comment by KristofferC** [comment]
Works fine here.

```
➜  Tokenize git:(master) ~/julia/julia

julia> versioninfo()
Julia Version 0.7.0-DEV.4321
Commit 7d46e56d3b* (2018-02-20 22:10 UTC)

julia> using Tokenize
[ Info: Precompiling module Tokenize

julia> Pkg.status("Tokenize")
WARNING: Base.Pkg is deprecated, run `using Pkg` instead
 in module Main
 - Tokenize                      0.4.2+             master

julia> collect(tokenize("function f(x) end"))
9-element Array{Tokenize.Tokens.Token,1}:
 1,1-1,8          KEYWORD        "function"
 1,9-1,9          WHITESPACE     " "
 1,10-1,10        IDENTIFIER     "f"
 1,11-1,11        LPAREN         "("
 1,12-1,12        IDENTIFIER     "x"
 1,13-1,13        RPAREN         ")"
 1,14-1,14        WHITESPACE     " "
 1,15-1,17        KEYWORD        "end"
 1,18-1,17        ENDMARKER      ""
```

**Comment by MikeInnes** [comment]
It would be great to have a new tag, as things are working well with Juno on the current master.

**Comment by MikeInnes** [comment]
It would be great to have a new tag, as things are working well with Juno on the current master.

**Comment by KristofferC** [comment]
Sure, made a tag.

**Comment by Keno** [comment]
Fixed by #129
